debezium-core/src/test/java/io/debezium/data/VerifyRecord.java

```diff
@@ -1090,7 +1090,11 @@ private static boolean areConnectSchemasEqual(Schema schema1, Schema schema2) {
             valueSchemasEqual = Objects.equals(schema1.valueSchema(), schema2.valueSchema());
         }
         else if (schema1.type() == Type.ARRAY && schema2.type() == Type.ARRAY) {
-            valueSchemasEqual = Objects.equals(schema1.valueSchema(), schema2.valueSchema());
+            // TODO:
+            // valueSchemasEqual = Objects.equals(schema1.valueSchema(), schema2.valueSchema());
```

Comment:
This is returning `false` for some reason, but I'm not sure why. The values appear to be the same when I print them:
```
valueSchemasEqual: false
SCHEMA 1: {"name" : "io.debezium.data.Enum", "type" : "STRING", "optional" : "false", "version" : "1"}
SCHEMA 2: {"name" : "io.debezium.data.Enum", "type" : "STRING", "optional" : "false", "version" : "1"}
```

Any ideas?

Reply:
Correct me if I'm wrong, but I _think_ the issue here is that this testing code assumes the objects for schema1 and schema2 will be the same. We're comparing by reference instead of value. I _think_ this has been okay so far because other tests that hit this branch use a shared constant like `SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA)`. This isn't true for my case, so this falls apart. I think I just have to make this comparison logic compare by value rather than reference.

Reply:
I imagine there might be a better way of doing this, but I'll start with this and see what suggestions others might have.
https://github.com/debezium/debezium/pull/1454/commits/c9c8b0a0a46331b208d874a59c935a5cfdb50478

Reply:
I think it failed because the expected value for the UPDATE case was not correct: the schema parameters should still be V1,V2, also after the UPDATE, as it only affects the actual column value, but not the allowed enum constants. I've pushed a commit for adjusting this.